[agent] Update nodeAffinity keys to avoid warnings

marojor opened the pull request:
```
root@qa-netsec-load:/home/ubuntu# kubectl -n sysdig-agent patch ds sysdig-agent --type='json' -p='[{"op": "add", "path": "/spec/template/spec/containers/0/image", "value": "sysdig/agent-kmodule:12.9.1"}]'
Warning: spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[1].matchExpressions[0].key: beta.kubernetes.io/arch is deprecated since v1.14; use "kubernetes.io/arch" instead
Warning: spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[1].matchExpressions[1].key: beta.kubernetes.io/os is deprecated since v1.14; use "kubernetes.io/os" instead
daemonset.apps/sysdig-agent patched
```

marojor committed:
[agent] Update nodeAffinity keys to avoid warnings

```
root@qa-netsec-load:/home/ubuntu# kubectl -n sysdig-agent patch ds sysdig-agent --type='json' -p='[{"op": "add", "path": "/spec/template/spec/containers/0/image", "value": "sysdig/agent-kmodule:12.9.1"}]'
Warning: spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[1].matchExpressions[0].key: beta.kubernetes.io/arch is deprecated since v1.14; use "kubernetes.io/arch" instead
Warning: spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[1].matchExpressions[1].key: beta.kubernetes.io/os is deprecated since v1.14; use "kubernetes.io/os" instead
daemonset.apps/sysdig-agent patched
```

aroberts87 commented:
Hey @marojor! Thank you for the contribution! Please make sure to bump the chart versions of `charts/agent/Charts.yaml` and `charts/sysdig-deploy/Charts.yaml` and update `charts/agent/CHANGELOG.md` and `charts/sysdig-deploy/CHANGELOG.md` as documented in the pull request template: https://github.com/sysdiglabs/charts/blob/master/.github/PULL_REQUEST_TEMPLATE.md. Other than that things look good. 👍🏻

aroberts87 committed:
Merge branch 'master' into SMAGENT-4244

aroberts87 committed:
Update Agent Chart Version and Changelog

Bump the agent chart version and add entry into
the changelog for the removal of `beta` from
for the kubernetes.io/arch and kubernetes.io/os
nodeAffinity keys due to deprecation.
